### PR TITLE
Fix SQL SUM & AVG aggregations

### PR DESCRIPTION
**Pull request by gierlachg**
Simplified `HazelcastSumAggFunction` & `HazelcastAggFunction` - operand types are no longer widened to result types.
Changed the result type of SUM(BIGINT) from `BIGINT` to `DECIMAL`.

### CODE REVIEW

File: hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/aggregate/function/HazelcastSumAggFunction.java [hunk 1]
```diff
@@ -64,9 +64,9 @@ protected boolean checkOperandTypes(HazelcastCallBinding binding, boolean throwO
             case TINYINT:
             case SMALLINT:
             case INTEGER:
+            case BIGINT:
```
Comment (hakanmemisoglu) [inline hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/aggregate/function/HazelcastSumAggFunction.java:67]: Isn't the previous version safer than new one?

**Review by hakanmemisoglu — APPROVED** [review]
LGTM, just minor question.

File: hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/aggregate/function/HazelcastSumAggFunction.java [hunk 2]
```diff
@@ -64,9 +64,9 @@ protected boolean checkOperandTypes(HazelcastCallBinding binding, boolean throwO
...
                 resultTypeName = BIGINT;
                 break;
-            case BIGINT:
```
Comment (viliam-durina) [inline hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/aggregate/function/HazelcastSumAggFunction.java:69]: It's done the wrong way. SUM(BIGINT) should result to DECIMAL, as in PG.

Comment (viliam-durina) [inline hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/aggregate/function/HazelcastSumAggFunction.java:69]: It's done the wrong way. SUM(BIGINT) should result to DECIMAL, as in PG.

**Review by viliam-durina — CHANGES_REQUESTED** [review]

**Review by viliam-durina — APPROVED** [review]

**Pull request merged**